drivers/soundwire/intel.c

```diff
@@ -424,6 +425,12 @@ static int intel_link_power_down(struct sdw_intel *sdw)
 	link_control &=  spa_mask;
 
 	ret = intel_clear_bit(shim, SDW_SHIM_LCTL, link_control, cpa_mask);
+	while (ret == -EAGAIN && retry >= 0) {
+		msleep(1);
+		ret = intel_clear_bit(shim, SDW_SHIM_LCTL,
+				      link_control, cpa_mask);
+		retry--;
+	}
```

Comment:
@bardliao intel_clear_bit() seems to be trying 10 times already? How will this help? Are we possibly missing some kind of lock/mutex maybe?

Reply:
@ranj063 It is for https://github.com/thesofproject/linux/issues/1555. Indeed, there may be some other issues. I will check it. Thanks.

Reply:
@ranj063 You are right. According to spec, we should wait for CPA to match SPA before we can program SPA again (https://github.com/thesofproject/linux/pull/1573). On the other hand, @plbossart should we just print a warning and don't return error when link power down failed? 
ie.
```
	ret = intel_link_power_down(sdw);
	if (ret) {
-		dev_err(dev, "Link power down failed: %d", ret);
-		return ret;
+		dev_warn(dev, "Link power down failed: %d", ret);
	}

```
The only affect is power consumption, right?

Reply:
@bardliao it'd prefer it if we root cause this instead of removing an error code....

To me it's either that the IP is already powered down, or that somehow we have a conflict between accesses to this register.